include/core/UnwrapRef.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+#ifndef INCLUDE_CORE_UNWRAPREF_H_
+#define INCLUDE_CORE_UNWRAPREF_H_
+
+#include <functional>
+
+namespace ml {
```

Comment:
Please can you add a Doxygen comment.  Something along the lines of:

```
//! Temporary substitute for std::unwrap_ref. This will be replaced by
//! std::unwrap_ref once all our compilers support it. The names in this
//! file do not conform to the coding style to ease the eventual transition
//! to std::unwrap_ref.
```